### PR TITLE
Fixes #575 - Implement Internationalization

### LINKED ISSUE #575: Propose and implement internationalization support
What options do we have for the Internationalization of the UI? Please propose and implement it. Please ensure settings are stored on the server.

### PR DESCRIPTION
**Pull request by rishiraj824**
Fixes issue #575 

Changes: 
- Added Internationalization to TopBar component, changes in the screenshot.
- Currently the German is supported, will add the support with the help of all codevs

Head to Settings, change the default language .

Demo Link:

http://susi-international.surge.sh

Screenshots for the change: 

- Select Language in `/settings`

screenshot from 2017-08-11 22 52 41

- Top bar component changes the content based on strings.js

screenshot from 2017-08-11 22 57 42

### CODE REVIEW

File: src/components/Translate/Translate.react.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+componentDidMount() {
+
+		let defaultPrefLanguage = this.state.defaultPrefLanguage;
+		let text = this.state.text;
+		if(defaultPrefLanguage!=='en-US' || text!==null){
+      // Translate the message text
+        let urlForTranslate = 'https://translate.googleapis.com/translate_a/single?client=gtx&sl=en-US&tl='+defaultPrefLanguage+'&dt=t&q='+text;
+        $.ajax({
```
Comment (sudheesh001) [inline src/components/Translate/Translate.react.js]: The indentation seems to be a little off here.

**Comment by rishiraj824** [comment]
@sudheesh101 Yes totally agreed, the text isn't always correct as it is being fetched from Google Translate, although I have made the default text as the one to render when the language is `en-US` and for that, no Ajax will be called. Thanks for reviewing :smile: Fixed the indentation as you said !

**Comment by rishiraj824** [comment]
@mariobehling Yes I agree, but I didn't find anything better than Google Translate. 

> The complete UI needs to be translated to different languages.

But how do I translate without using any other service?

**Comment by mariobehling** [comment]
Yes, language files. Of course this is the way to go, e.g. po/pot files. What is the appropriate way for ReactJS projects?

**Comment by rishiraj824** [comment]
@mariobehling I am proceeding with the strings file. I found a good blog and a good package which I can use. Marking this as [WIP] for now. 
https://www.npmjs.com/package/react-localization
http://www.bebetterdeveloper.com/coding/getting-started-react-flux.html

**Comment by mariobehling** [comment]
@rishiraj824 What is the status?

**Comment by rishiraj824** [comment]
@mariobehling Sorry for the delay, I have generated a [strings.js](https://github.com/fossasia/chat.susi.ai/pull/712/files#diff-c93e7673c31dd9a63a8e43280c226529) file for getting languages translation based on this file. Although I need the help of @uday96 @madhavrathi @isuruAb to add the details accordingly for all the components.

**Comment by madhavrathi** [comment]
@rishiraj824 we will divide the components among us.

**Comment by rishiraj824** [comment]
@madhavrathi Yes, I'll create the issue for the same. It will be faster if we divide the work and the strings accordingly. Currently, it works for the TopBar component. We will add the strings from all other components.

**Comment by mariobehling** [comment]
Please ensure the translation of content works with systems such a Weblate (https://weblate.org).

**Comment by rishiraj824** [comment]
Okay will make sure :+1:

**Comment by rishiraj824** [comment]
@mariobehling I have added [po/pot files](https://github.com/fossasia/chat.susi.ai/pull/712/files#diff-196818c81f733c6e05069f5a9211cd61) in the app, once all the strings from the components are added, we can integrate the project with http://weblate.org. 
Weblate uses the following JSON format for Javascript projects. I have formatted our `locales` based on that format. Here's the documentation https://docs.weblate.org/en/latest/formats.html#json-and-nested-structure-json-files. Once these files are complete we can have a [dashboard](https://hosted.weblate.org/hosting) where we can submit the po files for review. You can check the translation working at http://susi-international.surge.sh by changing the language inside `/settings`

**Comment by mariobehling** [comment]
Ok, thank you. Please let me know when all strings are added.

**Comment by uday96** [comment]
@rishiraj824 Made a PR to your fork for adding pot files for settings, TTS settings and custom server (partly fixes - https://github.com/fossasia/chat.susi.ai/issues/729). Please have a look - rishiraj824/chat.susi.ai#1 .

**Comment by rishiraj824** [comment]
@mariobehling Please check we have added all the strings from all the components. http://susi-international.surge.sh

**Comment by mariobehling** [comment]
Ok, thanks. Let's proceed from here. Cheers.

File: src/components/Auth/ChangePassword/ChangePassword.react.js [hunk 2]
```diff
@@ -207,7 +208,7 @@ export default class ChangePassword extends Component {
                         state.success = true;
                     }
                     else{
-                      msg = response.message+'\n Please Try Again.';
+                      msg = response.message+ '\n Please Try Again.';
```
Comment (srungta) [inline src/components/Auth/ChangePassword/ChangePassword.react.js]: @rishiraj824 Do you think we need to translate "Please Try again" as well. Was this intentionally skipped?

Comment (sudheesh001) [reply src/components/Auth/ChangePassword/ChangePassword.react.js]: Good catch! Ideally this needs to be translated too. @rishiraj824 can you please take a look

Comment (rishiraj824) [reply src/components/Auth/ChangePassword/ChangePassword.react.js]: @sudheesh001 @srungta If you see all these messages are translated as well. Go down to where they are being rendered in the component you will find them being sent to the translate component. https://github.com/fossasia/chat.susi.ai/pull/712/files/f8511116a2e20e73dbb5bfd771f82e5c49a12dc2#diff-e4fbf5b6fceb1d5e5744aa918f9b2d40R321

**Pull request merged**